src/org/labkey/cds/data/steps/PopulateStudiesTask.java

```diff
@@ -135,6 +151,99 @@ private void cleanContainers(Set<String> importStudies, Container project, User
         logger.info("Deleted " + deleted + " studies in " + DateUtil.formatDuration(finish - start) + ".");
     }
 
+    private void ensurePermissions(Logger logger) throws PipelineJobException
+    {
+        QuerySchema schema = DefaultSchema.get(user, project).getSchema("cds");
+
+        if (null == schema)
+            throw new PipelineJobException("Unable to find cds schema.");
+
+        TableInfo importStudyGroups = schema.getTable("import_studygroups");
+
+        if (null == importStudyGroups)
+            throw new PipelineJobException("Unable to find cds.import_studygroups table.");
+
+        SQLFragment sql = new SQLFragment("SELECT * FROM ").append(importStudyGroups);
+        Map<String, Object>[] importStudies = new SqlSelector(importStudyGroups.getSchema(), sql).getMapArray();
+
+        // Map((Study, Role) -> Set(Groups))
+        Map<Pair<String, String>, Set<String>> studyRoleGroups = new HashMap<>();
+
+        for (Map<String, Object> study : importStudies)
+        {
+            String prot = (String) study.get("prot");
+            String role = (String) study.get("role");
+            String group = (String) study.get("group");
+
+            Pair<String, String> key = new Pair<>(prot, role);
+            if (!studyRoleGroups.containsKey(key))
+            {
+                studyRoleGroups.put(key, new HashSet<>());
+            }
+
+            studyRoleGroups.get(key).add(group);
+        }
+
+        Map<String, Role> simpleNameToRole = new HashMap<>();
+        for (Role r : RoleManager.getAllRoles())
+        {
+            simpleNameToRole.put(r.getName(), r);
+        }
+
+        // Wipes out previous settings
+        for (Container c : ContainerManager.getAllChildren(project))
+        {
+            MutableSecurityPolicy policy = (new MutableSecurityPolicy(c));
+            SecurityPolicyManager.savePolicy(policy);
+        }
+
+        // Creates reference list of container names
+        List<String> allContainerNames = new ArrayList<>(ContainerManager.getAllChildren(project).size());
+        for (Container c : ContainerManager.getAllChildren(project))
+        {
+            if (!c.equals(project)) {
+                allContainerNames.add(c.getName());
+            }
+        }
+
+        for (Map.Entry<Pair<String, String>, Set<String>> entry : studyRoleGroups.entrySet())
+        {
+            Container c = ContainerManager.getChild(project, entry.getKey().getKey());
+            allContainerNames.remove(c.getName());
+            MutableSecurityPolicy policy = (new MutableSecurityPolicy(c));
```

Comment:
This line creates a blank policy for each entry, which is problematic as it erases role assignment from previous entries for the same study.